### PR TITLE
[dy] Add mage secrets

### PR DESCRIPTION
**Pull request by dy46**
# Summary

Support creating encrypted secrets within Mage. I originally tried to put the secrets in the variables tab of the sidekick, but they are actually quite different from variables, so I moved them to a separate tab.

Screenshot 2023-02-14 at 10 56 12 AM

# Tests
tested locally

cc: @wangxiaoyou1993

### CODE REVIEW

File: mage_ai/data_preparation/repo_manager.py [hunk 1]
```diff
@@ -120,3 +130,51 @@ def set_repo_path(repo_path: str) -> None:
 
 def get_variables_dir(repo_path: str = None) -> str:
     return get_repo_config(repo_path=repo_path).variables_dir
+
+
+def create_secret(name: str, value: str):
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/repo_manager.py:135]: Move the secrets related methods into a separate file. It doesn't seem that they should be in this repo_manager.py.

File: mage_ai/data_preparation/repo_manager.py [hunk 2]
```diff
@@ -120,3 +130,51 @@ def set_repo_path(repo_path: str) -> None:
...
+    from mage_ai.orchestration.db.models import Secret
+    secrets_dir = os.path.join(
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/repo_manager.py:137]: The secrets dir should not be in the repo (code)'s folder. We want to separate the code and the data. For example, in the [local to cloud](https://docs.mage.ai/production/ci-cd/local-cloud/repository-setup) mode, user deploys their prebuilt docker image to cloud, any change to the repo's folder won't be persisted on cloud.
We can put the secret key in data's folder.

File: mage_ai/data_preparation/shared/utils.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from mage_ai.data_preparation.repo_manager import get_secrets
+from typing import Callable, Dict
+import os
+
+
+def get_template_vars() -> Dict[str, Callable]:
+    kwargs = dict(
+        env_var=os.getenv,
+        mage_secret_var=lambda x: get_secrets().get(x),
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/shared/utils.py:9]: Getting all the secrets is too expensive. We can just query the secret by name.

**Comment by dy46** [comment]
@wangxiaoyou1993 addressed comments

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**